canisters/src/vts/src/lib.rs

```diff
@@ -56,3 +60,74 @@ fn upload_firmware(req: UploadFirmwareRequest) -> VTSResult<()> {
 
 // Enable Candid export (see https://internetcomputer.org/docs/current/developer-docs/backend/rust/generating-candid)
 ic_cdk::export_candid!();
+
+#[derive(CandidType, Deserialize, Debug)]
+pub struct Provider {
+    agreements: Vec<u128>,
+}
+
+#[derive(CandidType, Deserialize, Debug)]
+struct Customer {
+    agreements: Vec<u128>,
+}
+
+#[derive(Clone, Debug)]
+pub struct AgreementConditions {
+    pub daily_usage_fee: Decimal,
+    pub gas_price: Decimal,
+}
+
+#[derive(Clone, Debug)]
+pub struct Agreement {
+    pub name: String,
+    pub vh_provider: Principal,
+    pub vh_customer: Principal,
+    pub state: AgreementState,
+    pub conditions: AgreementConditions,
+    pub vehicles: Vec<u128>,
+}
+
+#[derive(Clone, CandidType, Deserialize, Debug, PartialEq)]
+pub enum AgreementState {
+    Unsigned,
+    Signed,
+}
+
+#[derive(Clone, Default)]
+pub struct CanisterState {
```

Comment:
I am not sure that it is good idea to keep all state inside one structure. Because when smart contract needs to read, it will read whole state, even you need jus agreement information, maybe it is better to split?